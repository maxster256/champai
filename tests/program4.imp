DECLARE
    a;
    sito(2:5);
IN
    a := 10;
    sito(2) := 5;

    FOR i FROM 15 DOWNTO 5
    DO
        WRITE i;
    ENDFOR
END
