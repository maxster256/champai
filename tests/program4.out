SUB A A
SUB H H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
STORE H
SUB A A
INC A
INC A
SUB H H
INC H
INC H
INC H
INC H
INC H
STORE H
SUB A A
INC A
INC A
INC A
INC A
INC A
INC A
SUB H H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
INC H
STORE H
SUB A A
INC A
INC A
INC A
INC A
INC A
INC A
LOAD F # GETVAL
COPY H F
SUB G G
INC G
INC G
INC G
INC G
INC G
INC F # BEGIN GE/LE COND
SUB F G
JZERO F 93 # END GE/LE COND
SUB A A
INC A
INC A
INC A
INC A
INC A
INC A
LOAD H
PUT H
SUB A A
INC A
INC A
INC A
INC A
INC A
INC A
LOAD F # GETVAL
COPY H F
DEC F # FOR_LOOP_ITER++
SUB A A
INC A
INC A
INC A
INC A
INC A
INC A
STORE F # STORE_VAL
JUMP 47 # FOR_LOOP_COND_CHECK
HALT